external/storm-elasticsearch/src/main/java/org/apache/storm/elasticsearch/bolt/ElasticSearchClient.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.elasticsearch.bolt;
+
+import java.io.Serializable;
+
+import org.apache.storm.elasticsearch.common.EsConfig;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.client.transport.TransportClient;
+import org.elasticsearch.common.settings.ImmutableSettings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+
+final class ElasticSearchClient implements Serializable {
+
+    private final EsConfig esConfig;
+
+    ElasticSearchClient(EsConfig esConfig) {
+        this.esConfig = esConfig;
+    }
+
+    Client construct() {
+        Settings settings = createBasicSettings();
+        TransportClient transportClient = new TransportClient(settings);
+        addTransportAddresses(transportClient);
+        return transportClient;
+    }
+
+    private Settings createBasicSettings() {
+        return ImmutableSettings.settingsBuilder()
+                                .put("cluster.name", esConfig.getClusterName())
+                                .put("client.transport.sniff", "true")
```

Comment:
@harshach Well I don't know, to be quite honest with you. That code was there already and I assumed the author had a reason to put it there which may or may not be the case of course.